src/nvme/mi.h

```diff
@@ -955,6 +955,44 @@ int nvme_mi_admin_xfer(nvme_mi_ctrl_t ctrl,
 		       off_t resp_data_offset,
 		       size_t *resp_data_size);
 
+/**
+ * nvme_mi_admin_admin_passthru() - Submit an nvme admin passthrough command
+ * @ctrl: Controller to send command to
+ * @opcode:	The nvme admin command to send
+ * @flags:	NVMe command flags (not used)
+ * @rsvd:	Reserved for future use
+ * @nsid:	Namespace identifier
+ * @cdw2:	Command dword 2
+ * @cdw3:	Command dword 3
+ * @cdw10:	Command dword 10
+ * @cdw11:	Command dword 11
+ * @cdw12:	Command dword 12
+ * @cdw13:	Command dword 13
+ * @cdw14:	Command dword 14
+ * @cdw15:	Command dword 15
+ * @data_len:	Length of the data transferred in this command in bytes
+ * @data:	Pointer to user address of the data buffer
+ * @metadata_len:Length of metadata transferred in this command(not used)
+ * @metadata:	Pointer to user address of the metadata buffer(not used)
```

Comment:
I understand these to be present to match the existing API. However, if we break that with the bidirectional support, we could remove these too.

Reply:
Ok. maybe @keithbusch can remember why the APIs are designed this way.

Reply:
IIRC, the API is following the IOCTL interface. So this might explain why the buffer is reused. If this it the case I don't think we should start trying to change this model in libnvme and hide this. It usually doesn't end to good if one tries to be clever.

Reply:
We may need to add new interface to handle bidirectional commands. Again, it is out of scope of this pull request.